conanfile.py

```diff
@@ -0,0 +1,126 @@
+
+    def _download_arrayfire(self):
+        self.af_installer_local_path = BINARY_INSTALLER_NAME
+        if not os.path.exists(self.af_installer_local_path):
+            self.output.info(
+                f"Downloading the ArrayFire {ARRAYFIRE_VERSION} binary installer...")
+            tools.download(
+                f"https://arrayfire.s3.amazonaws.com/{ARRAYFIRE_VERSION}/{BINARY_INSTALLER_NAME}", self.af_installer_local_path)
+            self.output.success(
+                f"ArrayFire {ARRAYFIRE_VERSION} binary installer successfully downloaded to {self.af_installer_local_path}")
+        else:
+            self.output.info(
+                f"ArrayFire {ARRAYFIRE_VERSION} binary installer already exists - skipping download.")
+
+    def _unpack_arrayfire(self):
+        if not os.path.exists(self.af_unpack_path):
+            os.mkdir(self.af_unpack_path)
+        self.output.info(
+            f"Unpacking ArrayFire {ARRAYFIRE_VERSION} binary installer...")
+        cmd = f"bash {self.af_installer_local_path} --prefix={self.af_unpack_path} --skip-license"
+        self.run(cmd)
+        self.output.success(
+            f"ArrayFire {ARRAYFIRE_VERSION} successfully unpacked.")
+
+    def _process_arrayfire(self):
+        # Install ArrayFire to requisite path
+        self.af_unpack_path = os.path.join(self.source_folder, 'arrayfire')
+
+        # Only proceed if missing
+        if os.path.exists(os.path.join(self.af_unpack_path, 'include', 'arrayfire.h')):
+            self.output.info(
+                f"ArrayFire {ARRAYFIRE_VERSION} already unpacked - skipping.")
+        else:
+            self._download_arrayfire()
+            self._unpack_arrayfire()
+
+    def build(self):
+        self._process_arrayfire()
+
+    def package(self):
+        # libs
+        self.copy("*.so", dst="lib", keep_path=False, symlinks=True)
+        self.copy("*.so.*", dst="lib", keep_path=False, symlinks=True)
+
+        # headers
+        self.copy("*.h", dst="include", src="arrayfire/include")
+        self.copy("*.hpp", dst="include", src="arrayfire/include")
+
+    def package_info(self):
+        self.cpp_info.libs = []
+        if self.options.unified_backend:
+            self.cpp_info.libs.extend([
+                f"libaf.so.{ARRAYFIRE_VERSION}",
+            ])
+        if self.options.graphics:
+            self.cpp_info.libs.extend([
+                "libforge.so.1.0.5",
+            ])
+        if self.options.cuda_backend:
+            self.cpp_info.libs.extend([
+                f"libafcuda.so.{ARRAYFIRE_VERSION}",
+                "libnvrtc-builtins.so",
+                f"libcudnn.so.{CUDA_TOOLKIT_VERSION}",
+                f"libcusparse.so.{CUDA_TOOLKIT_VERSION}",
+                f"libcublas.so.{CUDA_TOOLKIT_VERSION}",
+                f"libcusolver.so.{CUDA_TOOLKIT_VERSION}",
+                f"libnvrtc.so.{CUDA_TOOLKIT_VERSION}",
+                f"libcufft.so.{CUDA_TOOLKIT_VERSION}",
```

Comment:
Just a heads-up, with #2785 changes into master, next fix release onwards, the following libraries wouldn't be needed to be shipped. Thus, we can remove them for 3.7.2 or 3.8.0 onwards.
- cuFFT
- cuSolver
- cuSparse
- cuBlas

Reply:
Sounds good. 👍